### ISSUE TITLE
Use `score_name` as metric's key in Checkpoint

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#2129]
## 🚀 Feature

We can provide a simplification of our code when saving the best model by metric. 

Currently, to save 2 best models according to the validation accuracy, we are doing:
```python
best_model_handler = Checkpoint(
    {"model": model},
    save_handler,
    filename_prefix="best",
    n_saved=2,
    score_name="accuracy",
    score_function=Checkpoint.get_default_score_fn("accuracy"),
)
evaluator.add_event_handler(Events.COMPLETED, best_model_handler)
```

However, we can fetch define `score_function` as `Checkpoint.get_default_score_fn("accuracy")` by default if `score_name` is set. Thus, we can remove one line of code:
```python
best_model_handler = Checkpoint(
    {"model": model},
    save_handler,
    filename_prefix="best",
    n_saved=2,
    score_name="accuracy",
)
evaluator.add_event_handler(Events.COMPLETED, best_model_handler)
```

**Comment by KickItLikeShika** [comment]
Do we want to delete `score_function` argument?

**Comment by vfdev-5** [comment]
Definitely, not
